ultrasonic/src/main/java/org/moire/ultrasonic/service/CachedMusicService.kt

```diff
@@ -52,18 +52,21 @@ class CachedMusicService(private val musicService: MusicService) : MusicService
     private val activeServerProvider = inject(ActiveServerProvider::class.java)
     private val appDatabase = inject(AppDatabase::class.java).value
 
+    // Old style TimeLimitedCache
     private val cachedMusicDirectories: LRUCache<String, TimeLimitedCache<MusicDirectory?>>
     private val cachedArtist: LRUCache<String, TimeLimitedCache<MusicDirectory?>>
     private val cachedAlbum: LRUCache<String, TimeLimitedCache<MusicDirectory?>>
     private val cachedUserInfo: LRUCache<String, TimeLimitedCache<UserInfo?>>
     private val cachedLicenseValid = TimeLimitedCache<Boolean>(120, TimeUnit.SECONDS)
     private val cachedIndexes = TimeLimitedCache<Indexes>(60 * 60, TimeUnit.SECONDS)
-    private val cachedArtists = TimeLimitedCache<Indexes>(60 * 60, TimeUnit.SECONDS)
     private val cachedPlaylists = TimeLimitedCache<List<Playlist>?>(3600, TimeUnit.SECONDS)
     private val cachedPodcastsChannels = TimeLimitedCache<List<PodcastsChannel>?>(3600, TimeUnit.SECONDS)
     private val cachedMusicFolders = TimeLimitedCache<List<MusicFolder>>(10 * 3600, TimeUnit.SECONDS)
     private val cachedGenres = TimeLimitedCache<List<Genre>>(10 * 3600, TimeUnit.SECONDS)
 
+    // New Room Database
+    private val cachedArtists =  appDatabase.artistsDao()
+
     private var restUrl: String? = null
```

Comment:
Because I used the same method calls in the Room DAO as in the TimeLimitedCache, changing the cache provide is a one liner..